upup/pkg/fi/cloudup/apply_cluster.go

```diff
@@ -575,17 +592,12 @@ func findHash(url string) (*hashing.Hash, error) {
 }
 
 // upgradeSpecs ensures that fields are fully populated / defaulted
-func (c *ApplyClusterCmd) upgradeSpecs() error {
+func (c *ApplyClusterCmd) upgradeSpecs(channel *api.Channel) error {
```

Comment:
Should the channel be in the `ApplyClusterCmd` here? Maybe add it as a top level member.. `c.channel` for instance..